_summaries/bahdanau2014neural_2.md

```diff
@@ -1,19 +1,70 @@
+
+Further, it uses a bi-directional RNN (BiRNN) as the encoder to summarise both the preceding and the following words in the encoding of each word.
+
+## How is it realized?
+
+The baseline RNN encoder-decoder encodes the input sequence $$\mathrm{x}=(x_1,\cdots,x_{T_x})$$ into a fixed-length vector $$c$$ using an RNN as follows:
+
+$$h_t=f(x_t,h_{t-1}), \qquad c=q(\{h_1,\cdots,h_{T_x}\})$$
+
+where $$h_t$$ is the hidden state at time $$t$$ and $$f(\cdot)$$ and $$q(\cdot)$$ are some nonlinear functions. The decoder part of the architecture uses another RNN which is trained to predict the probability of the next word at time $$t$$ using another RNN as some nonlinear function $$g(\cdot)$$ of the previous word $$y_{t-1}$$, the fixed-length encoding $$c$$ and the current state of the RNN $$s_t$$.
+
+The proposed architecture also uses a similar encoder as above, except that it uses two RNNs - one running in each direction to generate two sets of hidden states $$(\overrightarrow{h_1},\cdots,\overrightarrow{h_{T_x}})$$ and $$(\overleftarrow{h_1},\cdots,\overleftarrow{h_{T_x}})$$ for each input sequence.  The final hidden state (also called _annotation_) for each word is simply the concatenation of the forward and backward hidden states, i.e. $$h_{j}=\left [ \overrightarrow{h_j^T}; \overleftarrow{h_j^T} \right ]^T$$. This is to include information from both preceding and following words in the annotation of each word. Note that for the proposed model, $$j$$ and $$i$$ are used to index the input and output sequences respectively.
+
+<img width="250" alt="bahdanau2014neural_2a" src="bahdanau2014neural_2a.png">
+
+
+The paper largely modifies the decoder architecture of the baseline model where instead of having a fixed encoding $$c$$ for each sequence pair, there is an independent **context-vector $$c_{i}$$** for predicting each word in the output sequence. So, $$c$$ is replaced by $$c_{i}$$ in all equations for predicting the next word $$y_{i}$$ and computing the next state $$s_{i}$$ of the decoder RNN. 
+
+As earlier, the encoding vector(s) are computed from the hidden states generated by the encoder. However, here the hidden states are weighted by the parameters of an attention layer $$\alpha_{ij}$$:
+
+$$c_{i} = \sum_{j=1}^{T_x} \alpha_{i j} h_{j} \qquad \boldsymbol{\alpha}_i = softmax(\mathbf{e}_{i}) \qquad e_{ij} = a(s_{i-1},h_{j})$$
+
+where the last equation represents the parameters of an _alignment model_ that learns to assign an importance to inputs around $$j$$ for the output at $$i$$. This model $$a(\cdot)$$ is jointly trained with the other parts of the encoder-decoder architecture. This allows the model to search for positions in the input sequence with the most relevant information (captured in the context-vectors) for predicting next word.
+
+
+
+## Benefits of proposed model
```

Comment:
Perhaps reference some of the qualitative analysis around the words alignment or long sentence translations here.

Reply:
Sure, I can do that in brief. I was just worried about the length of the summary but I can refer it in a sentence. Thanks!